compiler/passes/normalize.cpp

```diff
@@ -694,6 +695,88 @@ static void insertCallTempsForRiSpecs(BaseAST* base) {
   }
 }
 
+
+class LowerThunkPrimsVisitor final : public AstVisitorTraverse
+{
+  public:
+    LowerThunkPrimsVisitor()          = default;
+   ~LowerThunkPrimsVisitor() override = default;
+
+    bool enterCallExpr(CallExpr* node) override;
+};
+
+static void collectThunkOuterVars(Expr* expr, std::set<Symbol*>& outerVars) {
+  std::vector<SymExpr*> uses;
+  collectSymExprs(expr, uses);
+  for (auto use : uses) {
+    Symbol* sym = use->symbol();
+    if (considerForOuter(sym))
+      outerVars.insert(sym);
+  }
+}
+
+static CallExpr* buildThunkPrimFunctions(CallExpr* node) {
+  CHPL_ASSERT(node->numActuals() == 1);
+  auto delayedExpr = node->get(1);
+
+  std::set<Symbol*> outerVars;
+  collectThunkOuterVars(delayedExpr, outerVars);
+
+  static int thunkUid = 0;
+  thunkUid++;
+  auto thunkUidStr = istr(thunkUid);
```

Comment:
Why did you need this enumeration/mangling here?

Reply:
I don't know that I strictly need it but it matches what we do with loop expressions etc. (each loop expression gets its own number).

Reply:
I see. OK, yeah. Admittedly, I relied on those enumerations in the logs before (maybe IDs could have sufficed, though). In any case, let's keep it.

Resolving.